assets/src/edit-story/components/canvas/useCanvasSelectionCopyPaste.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import uuid from 'uuid/v4';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, renderToString } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+import useClipboardHandlers from '../../utils/useClipboardHandlers';
+import { getDefinitionForType } from '../../elements';
+
+const DOUBLE_DASH_ESCAPE = '_DOUBLEDASH_';
+
+/**
+ * @param {?Element} container
+ */
+function useCanvasSelectionCopyPaste( container ) {
+	const {
+		state: { currentPage, selectedElements },
+		actions: { appendElementToCurrentPage, deleteSelectedElements },
+	} = useStory();
+
+	const copyCutHandler = useCallback(
+		( evt ) => {
+			const { type: eventType, clipboardData } = evt;
+
+			if ( selectedElements.length === 0 ) {
+				return;
+			}
+
+			const payload = {
+				sentinel: 'story-elements',
+				// @todo: Ensure that there's no unserializable data here. The easiest
+				// would be to keep all serializable data together and all non-serializable
+				// in a separate property.
+				items: selectedElements.map( ( element ) => ( {
+					...element,
+					basedOn: element.id,
+					id: undefined,
+				} ) ),
+			};
+			const serializedPayload = JSON.stringify( payload ).replace( /\-\-/g, DOUBLE_DASH_ESCAPE );
+
+			const textContent = selectedElements
+				.map( ( { type, ...rest } ) => {
+					const { TextContent } = getDefinitionForType( type );
+					if ( TextContent ) {
+						return TextContent( { ...rest } );
+					}
+					return type;
+				} )
+				.join( '\n' );
+
+			const htmlContent = selectedElements
+				.map( ( { type, ...rest } ) => {
+					// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+					const { Save } = getDefinitionForType( type );
+					return renderToString( <Save { ...rest } /> );
+				} )
+				.join( '\n' );
+
+			clipboardData.setData( 'text/plain', textContent );
```

Comment:
Does this need separate handling for IE due to IE accepting only `Text` (e.g. `clipboardData.setData ( 'Text', input.value )`?

Reply:
For reference, the code from the legacy editor: https://github.com/ampproject/amp-wp/blob/eba2f3b1cd42f81b7e2b7d4687d74a5e606603ba/assets/src/stories-editor/blocks/amp-story-page/copy-paste-handler.js#L48-L65

This begs the question of which browsers we want to support. Perhaps we don't want to support IE 11 at all? cc @pbakaus

Reply:
Let's drop IE11. We will only support last two versions of Chrome, Safari, Edge and Firefox (we will try to cover 90%+ market that way)